compiler/rustc_error_codes/src/error_codes/E0308.md

```diff
@@ -3,16 +3,19 @@ Expected type did not match the received type.
 Erroneous code example:
 
 ```compile_fail,E0308
-let x: i32 = "I am not a number!";
-//     ~~~   ~~~~~~~~~~~~~~~~~~~~
-//      |             |
-//      |    initializing expression;
-//      |    compiler infers type `&str`
-//      |
-//    type `i32` assigned to variable `x`
+fn plus_one(x: i32) -> i32 {
+    x + 1
+}
+
+plus_one("Not a number");
+//       ^^^^^^^^^^^^^^ expected `i32`, found `&str`
+
+if "Not a bool" {
+// ^^^^^^^^^^^^ expected `bool`, found `&str`
+}
```

Comment:
Could you also add something similar to the original example? That way there's an example that explicitly shows a type mismatch as well.